src/ucp/wireup/wireup_cm.c

```diff
@@ -467,21 +651,33 @@ static ssize_t ucp_cm_server_priv_pack_cb(void *arg, const char *dev_name,
 free_addr:
     ucs_free(ucp_addr);
 out:
-    if (status != UCS_OK) {
-        wireup_ep = (ucp_wireup_ep_t *)ep->uct_eps[key.wireup_lane];
-        ucp_worker_set_ep_failed(worker, ep, &wireup_ep->super.super,
-                                 key.wireup_lane, status);
+    if (status == UCS_OK) {
+        ep->flags |= UCP_EP_FLAG_LOCAL_CONNECTED;
```

Comment:
isn't this too early to set this flag? should happen after getting connect ack from server? 
is it ok to send disconnect after this? because this flag decides whether to send disconnect or not..